src/opers.c

```diff
@@ -688,6 +694,12 @@ Obj FuncAND_FLAGS (
         // smaller.
         if ( flags1 < flags2 ) {
             flagsX = flags2;
+#           ifdef HPCGAP
+                if (!PreThreadCreation) {
```

Comment:
Yeah I am not a big fan either, but (a) I tried to match the existing code (if you look above/below, you'll see it already uses this style), and (b) in this particular case, with los of nested #ifs and #ifdefs, it actually felt somewhat helpful regarding the code readability.

That said, I am willing to change this. But perhaps it would be better to merge this as-is, and then overhaul this whole function -- indeed, what bothers me more than the indentation of these `#ifdef`s is their sheer number... Ah, if we were using C++, we could use a "LockHold" object which automatically releases the lock when going out of scope, removing the need for most of those ifdefs (namely all containing a `HashUnlock`).

I actually thought about a way to "fake" this in C: We split `FuncAND_FLAGS` into two functions, an "inner" one, which does the bulk of the work, and takes the lock; and an outer one, which calls the inner function, and after it returned, releases any locks taken by the inner one. Roughly like this:
```C
Obj AND_FLAGS_intern(Obj flags1, Obj flags2, Obj *locked)
{
...
                    *locked = flags1;
                    HashLock(*locked);
...
  // no HashUnlock necessary
  return ...
}

Obj FuncAND_FLAGS(Obj self, Obj flags1, Obj flags2)
{
  Obj locked = 0;
  Obj result = AND_FLAGS_intern(flags1, flags2, &locked);
#if defined(HPCGAP) && defined(AND_FLAGS_HASH_SIZE)
  if (locked)
    HashUnlock(locked);
# endif
  return result;
}
```
But this is a bigger change, so I didn't want to put it into this syncing, which is supposed to do minimal work.